src/axolotl/integrations/kernels/adapters/glm_moe_dsa.py

```
@@ -0,0 +1,213 @@
+    except Exception:  # pylint: disable=broad-except
+        pass
+
+
+def is_glm_moe_dsa_nvfp4_modelopt(cfg) -> bool:
+    """True iff the base model is a GLM-5.2 (``glm_moe_dsa``) NVFP4-modelopt checkpoint
+    (``quant_method=modelopt``, ``quant_algo=NVFP4``). Any failure returns False."""
+    try:
+        from transformers import AutoConfig
+
+        hf_cfg = AutoConfig.from_pretrained(cfg.base_model, trust_remote_code=True)
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)